### ISSUE TITLE
desurium not working with boost-1.52

### ISSUE THREAD

**Issue by TheSiege** [desura/Desurium#348]
desurium is currently not working with boost-1.52 as the main windows does not show up and the process crashes;

```
*** Caught unhandled unknown exception; terminating
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::thread_resource_error> >'
  what():  boost thread: thread not joinable: Invalid argument
```

**Comment by karolherbst** [comment]
from the boost 1.52 rel notes:
"Breaking changes when BOOST_THREAD_VERSION==3 (Default value since Boost 1.53): There are some new features which share the same interface but with different behavior. These breaking features are provided by default when BOOST_THREAD_VERSION is 3, but the user can however choose the version 2 behavior by defining the corresponding macro. As for the deprecated features, these broken features will be only available until boost 1.55."

maybe this will fix it:

``` patch
diff --git a/cmake/platform/unix/platform.cmake b/cmake/platform/unix/platform.cmake
index 736f012..e19e836 100644
--- a/cmake/platform/unix/platform.cmake
+++ b/cmake/platform/unix/platform.cmake
@@ -4,6 +4,7 @@

add_definitions(
   -DBOOST_FILESYSTEM_VERSION=3
+  -DBOOST_THREAD_VERSION=2
   -DBOOST_NO_SCOPED_ENUMS=1
   -D_LARGE_FILES
   -D_FILE_OFFSET_BITS=64
```

**Comment by TheSiege** [comment]
nah, seems to not work

**Comment by TheSiege** [comment]
since @karolherbst wants proof; I downgraded to boost-1.51 (which has no threads use flag) and desura works again, so it is definitly because of boost-1.52

**Comment by karolherbst** [comment]
I will try to fix this an add this to the 0.8 release, because it is currently very critical

**Comment by karolherbst** [comment]
see #354
